gse-demo/src/main/java/com/powsybl/gse/demo/GseDemo.java

```diff
@@ -9,18 +9,102 @@
+            //to log the rootThrowable to fix the root cause.
+        }
+        try {
+            //Try a file in the root directory, next to the executable
+            //The name in alphabetical order should be next to the executable.
+            PrintWriter printWriterRoot = new PrintWriter("../" + STARTUP_ERROR_LOGFILE_NAME);
+            t.printStackTrace(printWriterRoot);
+            printWriterRoot.flush();
+        } catch (Throwable throwableFileRoot) {
+            //Swallow throwableFileRoot here, we don't want to confuse the user.
+            //This is such a broken case that we don't want to
+            //log throwableFileRoot, we want the other mechanisms in this method
+            //to log the rootThrowable to fix the root cause.
+        }
+    }
+
+    //Using javapackager from jdk8,
+    //we must avoid throwing an exception at all costs because
+    //the javapackager launcher just prints an InvocationException
+    //message on stdout. This is the message you get, it has no information:
+    //  GseDemo Error invoking method.
+    //  GseDemo Failed to launch JVM
+    //
+    //Also stdout is not always available, most notably when a user
+    //launches the application by double clicking in a file explorer
     public static void main(String[] args) {
-        SLF4JBridgeHandler.removeHandlersForRootLogger();
-        SLF4JBridgeHandler.install();
+        //Don't use a static logger so that this method can run
+        //even if there is a problem loading the logger classes
+        Logger logger = null;
+        try {
+            logger = LoggerFactory.getLogger(GseDemo.class);
+
+            SLF4JBridgeHandler.removeHandlersForRootLogger();
+            SLF4JBridgeHandler.install();
 
-        GseApp.main(args);
+            GseApp.main(args);
+        } catch (Throwable rootThrowable) {
+            try {
+                if (logger != null) {
+                    try {
+                        logger.error(UNCAUGHT_EXCEPTION_MESSAGE, rootThrowable);
+                    } catch (Throwable slf4jThrowable) {
+                        //Swallow slf4jThrowable here, we don't want to confuse the user
+                        //This is such a broken case that we don't want to
+                        //log slf4jThrowable to fix slf4j, we want to log rootThrowable to fix the root cause.
+                        lastResortFallbackErrorLog(rootThrowable);
+                    }
+                } else {
+                    lastResortFallbackErrorLog(rootThrowable);
+                }
+            } finally {
+                //We always rethrow the original exception to let the java launcher do something we it.
+                //With javapackager from jdk8, this displays somewhat useless message on stdout and exits with code 1.
+                //On windows this opens a popup displaying the error messages.
+                throw rootThrowable;
```

Comment:
You should not throw from the finally block... Sonar considers that it's a [bug](https://sonarcloud.io/project/issues?id=com.powsybl%3Apowsybl-gse&pullRequest=198&resolved=false&types=BUG).

I think this is equivalent:
```
try {
} catch (Throwable rootThrowable) {
    if (logger != null) { ... }
    else { ... }
    throw rootThrowable;
}
```
If an exception is thrown by `lastResortFallbackErrorLog()` which exception shoud you throw?

Reply:
the difference between what you wrote and throwing in finally is precisely that if lastResortFallbackErrorLog throws, the root throwable is rethrown.

I argue that this is the behavior we want, because we don't want the user to fix the logging of root throwable, we want the user to fix the cause of the root throwable.